ForestDB: db.getExistingView(viewName) returns null for views created in previous app launch

As the title says, 
Database.getExistingView() returns null for views that were created in previous launches of the app.
The view's files exist on storage, but forestDB doesn't retrieve them in future launches, so I'm forced to recreate all views at every launch. 
Am I missing something here?
Is this the expected behavior? 
Should I re-create the views at every launch?

I'm using:
    compile 'com.couchbase.lite:couchbase-lite-android:1.3.0-45'
    compile 'com.couchbase.lite:couchbase-lite-android-forestdb:1.3.0-45'

Hi @hideki, 
Couldn't compile from sources, using the following submodules:
```
 8ab623904639f27575c715b03ca9597b88ed59a2 libraries/couchbase-lite-java-core (1.3.0-3-g8ab6239)
 edd741e46adc9779f8a6702869e8450b9880cc0c libraries/couchbase-lite-java-forestdb (1.3.0-7-gedd741e)
 501be320fea1396e5922c1ddbde08af4d2e87664 libraries/couchbase-lite-java-native (1.3.0)
```
NDK doesn't want do compile some file:
```
:libraries:couchbase-lite-java-native:sqlcipher:transformNative_libsWithSyncJniLibsForRelease UP-TO-DATE
:libraries:couchbase-lite-java-native:sqlcipher:bundleRelease
In file included from /home/victor/Documents/software/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/jni/../vendor/cbforest/vendor/forestdb/src/blockcache.cc:31:
In file included from ../vendor/cbforest/vendor/forestdb/src/blockcache.h:21:
In file included from ../vendor/cbforest/vendor/forestdb/src/filemgr.h:35:
In file included from ../vendor/cbforest/vendor/forestdb/src/atomic.h:30:
In file included from /home/victor/android-sdks/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/include/atomic:41:
/home/victor/android-sdks/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/atomic_base.h:443:16: error: cannot compile this atomic library call yet
      { return __atomic_and_fetch(&_M_i, __i, memory_order_seq_cst); }
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/victor/android-sdks/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/atomic_base.h:451:16: error: cannot compile this atomic library call yet
      { return __atomic_or_fetch(&_M_i, __i, memory_order_seq_cst); }
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.

/home/victor/android-sdks/ndk-bundle/build/core/build-binary.mk:462: recipe for target '/home/victor/Documents/software/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/obj/local/armeabi/objs/CouchbaseLiteJavaForestDB/__/vendor/cbforest/vendor/forestdb/src/blockcache.o' failed

make: *** [/home/victor/Documents/software/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/obj/local/armeabi/objs/CouchbaseLiteJavaForestDB/__/vendor/cbforest/vendor/forestdb/src/blockcache.o] Error 1

make: Leaving directory '/home/victor/Documents/software/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/jni'
:libraries:couchbase-lite-java-forestdb:buildNative FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':libraries:couchbase-lite-java-forestdb:buildNative'.
> Process 'command '/home/victor/android-sdks/ndk-bundle/ndk-build'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.779 secs

```

Anyway I looked deeper into ForestDBViewStore and have some interesting findings,

![screenshot from 2016-07-31 07 50 59](https://cloud.githubusercontent.com/assets/8295627/17276072/d88993be-56f3-11e6-9a88-0ec8d17009d8.png)
In this case, it is trying to open the file:
`/storage/emulated/0/Android/data/com.precisagro.agcampofree/files/dbfc3bdb2e3cb5826a46305d08baafe044.cblite2/users%2Famostragens5.viewindex`
However, inspecting the application's path, the file that actually exists is `users%2Famostragens5.viewindex.2`,
In fact for every viewname there is a .meta file and a .x file (x being an integer from 0 to 5).
By not finding the file, the method throws CouchbaseLiteException(404).
Looks like viewNameToFileName is computing the proper filename, but is still missing that last integer, that I've no idea from where it comes and what is means.

Thanks a lot,

--- a/src/androidTest/java/com/couchbase/lite/ViewsTest.java
+++ b/src/androidTest/java/com/couchbase/lite/ViewsTest.java
@@ -119,11 +119,10 @@ public class ViewsTest extends LiteTestCaseWithDB {
             db1.close();
 
             Database db2 = manager.getExistingDatabase(DEFAULT_TEST_DB);
-            putDocs(db2);
-            View view2 = db2.getView(viewName);
+            View view2 = db2.getExistingView(viewName);
             Assert.assertNotNull(view2);
-            assertEquals(expectedVersion,view2.getMapVersion());
-            assertEquals(Status.NOT_MODIFIED,view2.updateIndex());
+            //assertEquals(expectedVersion,view2.getMapVersion());
+            //assertEquals(Status.NOT_MODIFIED,view2.updateIndex());
             db2.close();
         } catch (CouchbaseLiteException e) {
             e.printStackTrace();

--- a/src/androidTest/java/com/couchbase/lite/ViewsTest.java
+++ b/src/androidTest/java/com/couchbase/lite/ViewsTest.java
@@ -98,6 +98,37 @@ public class ViewsTest extends LiteTestCaseWithDB {
 
         Assert.assertTrue(changed);
     }
+    
+    //https://github.com/couchbase/couchbase-lite-android/issues/952
+    public void testViewCreationAndRecall() {
+        try {
+            String expectedVersion = "1.0";
+            String viewName = "test/view";
+            Database db1 = startDatabase();
+            putDocs(db1);
+            View view1 = db1.getView(viewName);
+            view1.setMap(new Mapper() {
+                @Override
+                public void map(Map<String, Object> document, Emitter emitter) {
+                    if (document.get("key") != null) {
+                        emitter.emit(document.get("key"), null);
+                    }
+                }
+            }, expectedVersion);
+            view1.updateIndex();
+            db1.close();
+
+            Database db2 = manager.getExistingDatabase(DEFAULT_TEST_DB);
+            putDocs(db2);
+            View view2 = db1.getView(viewName);
+            Assert.assertNotNull(view2);
+            assertEquals(expectedVersion,view2.getMapVersion());
+            assertEquals(Status.NOT_MODIFIED,view1.updateIndex());
+            db2.close();
+        } catch (CouchbaseLiteException e) {
+            e.printStackTrace();
+        }
+    }
 
     //https://github.com/couchbase/couchbase-lite-java-core/issues/219
     public void testDeleteView() {

--- a/src/androidTest/java/com/couchbase/lite/ViewsTest.java
+++ b/src/androidTest/java/com/couchbase/lite/ViewsTest.java
@@ -121,8 +121,15 @@ public class ViewsTest extends LiteTestCaseWithDB {
             Database db2 = manager.getExistingDatabase(DEFAULT_TEST_DB);
             View view2 = db2.getExistingView(viewName);
             Assert.assertNotNull(view2);
-            //assertEquals(expectedVersion,view2.getMapVersion());
-            //assertEquals(Status.NOT_MODIFIED,view2.updateIndex());
+            view2.setMap(new Mapper() {
+                @Override
+                public void map(Map<String, Object> document, Emitter emitter) {
+                    if (document.get("key") != null) {
+                        emitter.emit(document.get("key"), null);
+                    }
+                }
+            }, expectedVersion);
+            assertEquals(Status.NOT_MODIFIED,view2.updateIndex().getCode());
             db2.close();
         } catch (CouchbaseLiteException e) {
             e.printStackTrace();

--- a/src/androidTest/java/com/couchbase/lite/ViewsTest.java
+++ b/src/androidTest/java/com/couchbase/lite/ViewsTest.java
@@ -120,10 +120,10 @@ public class ViewsTest extends LiteTestCaseWithDB {
 
             Database db2 = manager.getExistingDatabase(DEFAULT_TEST_DB);
             putDocs(db2);
-            View view2 = db1.getView(viewName);
+            View view2 = db2.getView(viewName);
             Assert.assertNotNull(view2);
             assertEquals(expectedVersion,view2.getMapVersion());
-            assertEquals(Status.NOT_MODIFIED,view1.updateIndex());
+            assertEquals(Status.NOT_MODIFIED,view2.updateIndex());
             db2.close();
         } catch (CouchbaseLiteException e) {
             e.printStackTrace();

